### ISSUE TITLE
ONE OF constraint doesn't handle booleans correctly

### ISSUE THREAD

**Issue by julie-sullivan** [intermine/intermine#1005]
From Alex

The null value constraint issue is a problem with the intermine pathquery engine. The tables are generating the correct query:

```
<query model="genomic" view="Gene.primaryIdentifier Gene.proteins.primaryAccession Gene.proteins.isUniprotCanonical" sortOrder="Gene.primaryIdentifier ASC" >
  <constraint path="Gene.proteins.isUniprotCanonical" op="ONE OF" >
    <nullValue/>
    <value>false</value>
  </constraint>
</query>
```

But the result returned do not include any nulls. Interestingly, on string columns that include nulls, the query engine behaves correctly.

See #641

**Comment by julie-sullivan** [comment]
@sergiocontrino Putting this on the next milestone but should be a point release.

**Comment by sergiocontrino** [comment]
i am not sure it is a bug and if there is a case for wanting to aggregate booleans with null/not null cases.
SQL itself deals the same way with booleans.
i.e. 
```
select id,isuniprotcanonical from protein where isuniprotcanonical in (NULL, false);
```
is the same as
```
select id,isuniprotcanonical from protein where isuniprotcanonical = false;
```
QB does not allow the offending query to be constructed, and i think we should extend this to the tables.
if we want to allow this request we need to transform the query using an OR instead of using the usual IN clause. in the example above
```
select id,isuniprotcanonical from protein where isuniprotcanonical is NULL or isuniprotcanonical=false;
```

bottom line: null should probably not be a condition for a boolean.

**Comment by julie-sullivan** [comment]
```
if we want to allow this request we need to transform the query using an OR instead of using the usual IN clause.
```
@sergiocontrino Where would that transformation happen?

**Comment by sergiocontrino** [comment]
i think that we should not allow the query on booleans, i.e. 'ONE OF'/'NONE OF'  should not be allowed on boolean fields. if for some reason it is important to have the functionality, we should transform the constraint into the OR i mentioned above. 
not sure where this would be more convenient, i can check if required.
currently it is not possible to build the query with the qb.

**Comment by sergiocontrino** [comment]
one of/none of are already invalid in imtable, as in the query builder. the only way of inputting the query mentioned in the first comment is via the 'import query from xml' tool, using hand-made xml.
@julie-sullivan i think we could close this. alternatively, we should not validate the xml (probably in https://github.com/intermine/intermine/blob/dev/intermine/pathquery/main/src/org/intermine/pathquery/PathQuery.java#1140?)
